algorithms/HPdutility/inst/tests/test_hpdsample.R

```diff
@@ -0,0 +1,180 @@
+    update(lvs)
+  }, progress = F)
+  locMeans <- getpartition(dlocMeans)
+  locVars  <- getpartition(dlocVars)
+
+#  print(paste('glob mean:', globalMean))
+#  print(paste('loc means:', paste(locMeans, collapse = ", ")))
+#  print(paste('glob var:', globalVar))
+#  print(paste('loc vars:', paste(locVars, collapse = ", ")))
+
+  expect_true(all((locMeans - globalMean)/globalMean < 0.1))
+  expect_true(all((as.vector(locVars) - globalVar)/globalVar < 0.1))
+})
+ 
+test_that("when data1 and data2 provided, samples correspond", {
+  d1 <- as.darray(as.matrix(1:1001))
+  d2 <- as.darray(as.matrix(1:1001) + 1)
+  nSamplePartitions <- 1:6
+  samplingRatios <- seq(0.1, 2, 0.5)
+  for (n in nSamplePartitions) {
+    for (sr in samplingRatios) {
+      sd <- hpdsample(d1, d2, nSamplePartitions = n, samplingRatio = sr)
+      s1 <- getpartition(sd[[1]])
+      s2 <- getpartition(sd[[2]])
+      expect_true(all(s1 + 1 == s2))
+    }
+  }
+})
+
+test_that("when data1 and data2 provided, works with all combo of dframe/darray", {
+  d1 <- as.darray(as.matrix(1:1001))
+  d2 <- as.dframe(as.data.frame(1:1001) + 1)
+  n  <- 6
+  sr <- 0.1
+  sd <- hpdsample(d1, d1, nSamplePartitions = n, samplingRatio = sr)
+  sd <- hpdsample(d2, d1, nSamplePartitions = n, samplingRatio = sr)
+  sd <- hpdsample(d1, d2, nSamplePartitions = n, samplingRatio = sr)
+  sd <- hpdsample(d2, d2, nSamplePartitions = n, samplingRatio = sr)
+})
+
+
+test_that("works with diff data types", {
+  d1 <- as.darray(as.matrix(1:1001))
+  d2 <- as.dframe(data.frame(a = rep('c', 1001), 
+                             b = 1:1001, 
+                             c = runif(1001), 
+                             d = c(rep('a', 501), rep('b', 500))))
+  print(str(getpartition(d2)))
+  n  <- 6
+  sr <- 0.1
+  # Shouldn't break
+  sd <- hpdsample(d1, d1, nSamplePartitions = n, samplingRatio = sr)
+
+  sd <- hpdsample(d2, nSamplePartitions = n, samplingRatio = sr)
+  expect_true(all(sapply(getpartition(sd), class) == 
+      sapply(getpartition(d2), class)))
+
+  sd <- hpdsample(d2, d1, nSamplePartitions = n, samplingRatio = sr)
+
+  # Make sure the classes of the output columns are the same as the ones of the
+  # input columns
+  expect_true(all(sapply(getpartition(sd[[1]]), class) == 
+      sapply(getpartition(d2), class)))
+  print(str(getpartition(sd[[1]])))
+
+  sd <- hpdsample(d1, d2, nSamplePartitions = n, samplingRatio = sr)
+  expect_true(all(sapply(getpartition(sd[[2]]), class) == 
+      sapply(getpartition(d2), class)))
+  sd <- hpdsample(d2, d2, nSamplePartitions = n, samplingRatio = sr)
+  expect_true(all(sapply(getpartition(sd[[2]]), class) == 
+      sapply(getpartition(d2), class)))
+})
+
+test_that("all output partitions have the same size", {
+  nSamplePartitions <- 1:5
+  for (n in nSamplePartitions) {
+    sa <- hpdsample(a, nSamplePartitions = n, samplingRatio = 0.5)
+    expect_true(all(partitionsize(sa)[,1] == partitionsize(sa)[1,1]))
+  }
+})
+
+
+distributedR_shutdown()
```

Comment:
Please remove "distributedR_shutdown()" as it is planned to be done out of the tests.